nav/virtual_layers/src/virtual_layer.cpp

```diff
@@ -0,0 +1,146 @@
+using costmap_2d::LETHAL_OBSTACLE;
+using costmap_2d::NO_INFORMATION;
+
+namespace virtual_layers
+{
+
+VirtualLayer::VirtualLayer() {
+  // ros::Time last_updated_ = ros::Time::now();
+  // ROS_INFO("Start");
+}
+
+void VirtualLayer::onInitialize()
+{
+  ros::NodeHandle nh("~/" + name_);
+  current_ = true;
+  default_value_ = NO_INFORMATION;
+  matchSize();
+
+  dsrv_ = new dynamic_reconfigure::Server<costmap_2d::GenericPluginConfig>(nh);
+  dynamic_reconfigure::Server<costmap_2d::GenericPluginConfig>::CallbackType cb = boost::bind(
+      &VirtualLayer::reconfigureCB, this, _1, _2);
+  dsrv_->setCallback(cb);
+}
+
+void VirtualLayer::matchSize()
+{
+  Costmap2D* master = layered_costmap_->getCostmap();
+  resizeMap(master->getSizeInCellsX(), master->getSizeInCellsY(), master->getResolution(),
+            master->getOriginX(), master->getOriginY());
+}
+
+
+void VirtualLayer::reconfigureCB(costmap_2d::GenericPluginConfig &config, uint32_t level)
+{
+  enabled_ = config.enabled;
+}
+
+std::vector<std::vector<double>> getRandomPoints(double t) {
+  // int n = 3;
+  // double point_x;
+  // double point_y;
+  std::vector<std::vector<double>> points = {{-1 - t , 1}, {-1 - t,-1}};
+  // std::random_device rd_x;
+  // std::random_device rd_y;
+  // std::mt19937 gen_x(rd_x());
+  // std::mt19937 gen_y(rd_y());
+  // std::uniform_real_distribution<double> rand_x(-4, 4);
+  // std::uniform_real_distribution<double> rand_y(-4, 4);
+
+  // for (int xy = 0; xy < n ; xy++){
+  //   point_x = rand_x(rd_x);
+  //   point_y = rand_y(rd_y);
+  //   points.push_back({point_x, point_y});
+  //   ROS_INFO("%f %f" , point_x, point_y);
+  // }
+
+  return points;
+}
+
+std::vector<std::vector<double>> pointsToLine(std::vector<std::vector<double>> points, double t) {
+  std::vector<std::vector<double>> coord;
+  coord.push_back({points[0][0], points[0][1]});
+  int m = points.size() - 1;
+  for (int j = 0; j < m; j++) {
+    double x1 = points[j][0];
+    double x2 = points[j+1][0];
+    double y1 = points[j][1];
+    double y2 = points[j+1][1];
+    double dis = sqrt(pow(x1 - x2, 2) + pow(y1 - y2, 2));
+    double alpha = t / dis;
+    for (double k = alpha; k <= 1; k += alpha){
+      double x_point = x1 + k * (x2 - x1);
+      double y_point = y1 + k * (y2 - y1);
+      coord.push_back({x_point, y_point});
+    }
+  } 
+  
+  return coord;
+}
+
+void VirtualLayer::updateBounds(double robot_x, double robot_y, double robot_yaw, double* min_x,
+                                           double* min_y, double* max_x, double* max_y)
+{
+  if (!enabled_)
+    return;
+  
+  if (initialize != true) {
+    initialize = true;
+  }
+
+  std::vector<std::vector<double>> points = getRandomPoints(counter);
+  std::vector<std::vector<double>> coord = pointsToLine(points, 0.05);
+  ROS_INFO("points");
+  
+  for (int i = 0; i < coord.size(); i++) {
+    double magnitude = sqrt(pow(coord[i][0],2) + pow(coord[i][1],2));
+    //double mark_x = magnitude*cos(robot_yaw) + robot_x, 
```

Comment:
You can probably remove the commented out code here 

You should make a note of it somewhere though to keep track of it just in case you want to switch to whatever it's doing